cypcore/Ledger/PosMinting.cs

```diff
@@ -75,122 +73,111 @@ public class PosMinting : IPosMinting
             _stakingConfigurationOptions = stakingConfigurationOptions;
             _logger = logger.ForContext("SourceContext", nameof(PosMinting));
             _keyPair = _signing.GetOrUpsertKeyName(_signing.DefaultSigningKeyName).GetAwaiter().GetResult();
-
-            _serialQueue = new SerialQueue();
-            _stakingTimer = new Timer(_ => Staking(), null, TimeSpan.FromSeconds(35), TimeSpan.FromSeconds(StakeTimeSlot));
-
-            applicationLifetime.ApplicationStopping.Register(OnApplicationStopping);
+            _timer = Observable.Timer(TimeSpan.FromSeconds(35), TimeSpan.FromSeconds(StakeTimeSlot)).Subscribe(_ =>
+            {
+                Stake().SafeFireAndForget(exception =>
+                {
+                    StakeRunning = false;
+                    _logger.Here().Error(exception, "Stake error");
+                });
+            });
         }
 
         public bool StakeRunning { get; private set; }
 
         /// <summary>
         /// 
         /// </summary>
-        private void OnApplicationStopping()
-        {
-            _logger.Here().Information("Application stopping");
-            _stakingTimer?.Change(Timeout.Infinite, 0);
-        }
-
-        /// <summary>
-        /// 
-        /// </summary>
-        /// <returns></returns>
-        private void Staking()
+        /// <exception cref="WarningException"></exception>
+        /// <exception cref="Exception"></exception>
+        private async Task Stake()
         {
             if (_sync.SyncRunning) return;
             if (StakeRunning) return;
 
-            async void TryStake()
+            try
             {
-                try
+                if (_stakingConfigurationOptions.OnOff != true)
```

Comment:
Just as a reminder to myself, we should rename this to "staking enabled". Staking OnOff == true is confusing.

Reply:
Simply staking would do as well.

Reply:
It's a field in a section already named "Staking": https://github.com/cypher-network/cypher/blob/master/cypnode/Configuration/Templates/appsettings.json#L53

Reply:
Then simply enabled would be just if not better.